### PR TITLE
Store spectral windows on the creation of a new technical goal

### LINKED ISSUE #12: spectral window not persisted
creating a spectral window, pressing "save", then pressing "finished editing". if you then open the technical goal again for editing, the window has disappeared.

### PR DESCRIPTION
**Pull request by DJWalker42**
Fixes #12.

- add a map of form spectral windows to convert to an appropriate type to save to the database

### CODE REVIEW

File: src/main/webui/src/technicalGoals/scienceSpectralWindowGui.tsx [hunk 1]
```diff
@@ -34,7 +34,7 @@ export type ExpectedSpectralLineGui = {
  *  Notice that we have transferred the members of 'SpectralWindowSetup'
  *  to be direct children of this type.
  *
- * @param {number | string} index optional id.
+ * @param {number | string} index unsure as to the semantics of this - has a non-null constraint in DB.
```
Comment (alan-stokes) [inline src/main/webui/src/technicalGoals/scienceSpectralWindowGui.tsx:37]: if this is anything like the NGOT system. a spectral window had both a group index and an index for 2 reasons, 1. was to identify the baseband it was in (from 1 to 4) and a index which stated where it was within the baseband window. This was partly due to the windows being stored as 1 array, and so you had to traverse the entire array to locate all windows for a given baseband and the same for the position within the baseband. Maybe this is something simular?

File: src/main/webui/src/technicalGoals/scienceSpectralWindowGui.tsx [hunk 2]
```diff
@@ -131,13 +131,13 @@ export function convertToExpectedSpectralLine(
...
  * @param {ScienceSpectralWindowGui} input the window gui to convert.
- * @return {}ScienceSpectralWindow} the converted window.
+ * @return {ScienceSpectralWindow} the converted window.
  */
 export function convertToScienceSpectralWindow(
         input: ScienceSpectralWindowGui): ScienceSpectralWindow {
     // return the new window
     return {
-        index: input.index as number,
+        index: Math.random() * 1000, //work-around: unclear on the semantics of 'index'
```
Comment (alan-stokes) [inline src/main/webui/src/technicalGoals/scienceSpectralWindowGui.tsx:140]: I personally dislike this view of adding a randomly generated number here. Why cant we just add a default thats consistent but not feasible. such as -1? then future us can see this as a regular value and not wonder why the number keeps changing?

**Review by alan-stokes — CHANGES_REQUESTED** [review]

**Comment by pahjbo** [comment]
will remove index from the model as we do not know what it is being used for.

**Comment by alan-stokes** [comment]
have been thinking about this a bit, and I think i have a answer which feels like it would explain things a bit, though i still think removing it if we control the database implementation is a good call, as it means less data to transmit and receive. 

its often a good practice to avoid implementation details of data storage. for example stating a collection of data as an Collection, and not a an arrayList, or a HashMap or a Set. as each of these implementations have different views on order. And so maybe someone felt to put a index in to ensure if the data was stored in, say a set, that they could still recover the order? It may also be a reaction to python hashmaps, which at some point did not retain order, but then got switched back to FIFO when iterating. 

Anyhow, just my thought of for why that index may exist.

**Pull request merged**